src/Utopia/Messaging/Adapter.php

```diff
@@ -75,14 +75,17 @@ protected function request(
         string $method,
         string $url,
         array $headers = [],
-        string $body = null,
+        string|array $body = null,
```

Comment:
What does it mean that the body can be either a string or any array, not sure if this is a great interface.

Reply:
Yeah I agree it's not great. Previously we only allowed string and forced the encoding (JSON or `http_build_query`) to be used from outside. 

Now we need to support multipart as well, which requires passing the parameters to CURL as an array, unless we do it the hard way and build the multipart body manually.

I can add some more checks and comments to make sure that arrays are only allowed if the content type if multipart and string for all others?

Reply:
Updated to force array only and encode internally depending on the content-type header